soc/arm/st_stm32/common/Kconfig.defconfig.series

```diff
@@ -24,9 +27,21 @@ config LOG_BACKEND_SWO_REF_FREQ_HZ
 endif # LOG_BACKEND_SWO
 
 # set the tick per sec as a divider of the LPTIM clock source
+if STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSE
 config SYS_CLOCK_TICKS_PER_SEC
-	default 4096 if STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSE
-	default 4000 if STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSI
+	default 4096 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" < 16
+	default 2048 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" = 16
+	default 1024 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" >= 32
+
+endif # STM32_LPTIM_TIMER
+
+if STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSI
```

Comment:
What about using `depends on STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSI` ?

Reply:
same result, changing to "depends on"